src/popup/utils/utils.ts

```diff
@@ -199,14 +199,15 @@ export function buildSimplexLink(address: string) {
 /**
  * Watch for the getter to be truthy with the use of the compositionApi.
  */
-export function watchUntilTruthy<T>(getter: () => T): Promise<NonNullable<T>> {
+export function watchUntilTruthy<T>(getter: WatchSource<T>): Promise<NonNullable<T>> {
   return new Promise((resolve) => {
     const unwatch = watch(
       getter,
       (value) => {
-        if (!value) return;
-        resolve(getter() as NonNullable<T>);
-        defer(() => unwatch());
+        if (value) {
+          resolve(value as NonNullable<T>);
+          defer(() => unwatch());
```

Comment:
```suggestion
          defer(unwatch);
```

Reply:
I would prefer to stick to what we have for the security reasons. Explained on private.